server/aesdsocket.c

```diff
@@ -15,16 +15,53 @@
 #include <sys/wait.h>
 #include <errno.h>
 #include <ctype.h>
+#include "queue.h"
+#include <pthread.h>
+#include <stdbool.h>
+#include <time.h>
+#include <sys/time.h>
 
 #define PORT_NUMBER "9000"
 #define BACKLOG_CONNECTIONS 6
-#define MAXDATASIZE 100 // max number of bytes we can get at once
+#define MAXDATASIZE 500 // max number of bytes we can get at once
 #define FILE_PATH "/var/tmp/aesdsocketdata"
+#define TIMER_BUFFER_SIZE 100
 
-int socket_fd, client_fd, write_file_fd;
+int socket_fd, client_fd, write_fd;
 struct addrinfo hints, *res;
 char ipstr[INET6_ADDRSTRLEN];
 uint8_t daemon_arg = 0;
+int terminate = 0;
+//slist_data_t *datap=NULL;
+
+
+// Used the following thread details struct from previous assignment
+typedef struct thread_data{
+
+    pthread_t thread_index;
+    int client_fd;
+    char *buffer;
+    char *transmit_buffer;
+    sigset_t signal_mask;
+    bool thread_complete_success;
+    pthread_mutex_t *mutex;
+}thread_data_t;
+
+pthread_mutex_t main_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
@Sharath-Jonnala what is this mutex locking?  Think about naming it based on what it's locking.